src/main/java/io/polivakha/mojo/properties/AbstractWritePropertiesMojo.java

```diff
@@ -24,118 +24,127 @@
+     * @param file       {@link File}
      * @throws MojoExecutionException {@link MojoExecutionException}
      */
-    protected void writeProperties( Properties properties, File file )
-        throws MojoExecutionException
-    {
-        try
-        {
-            storeWithoutTimestamp( properties, file, "Properties" );
-        }
-        catch ( FileNotFoundException e )
-        {
-            getLog().error( "Could not create FileOutputStream: " + file );
-            throw new MojoExecutionException( e.getMessage(), e );
-        }
-        catch ( IOException e )
-        {
-            getLog().error( "Error writing properties: " + file );
-            throw new MojoExecutionException( e.getMessage(), e );
+    protected void writeProperties(Properties properties, File file) throws MojoExecutionException {
+        try {
+            storeWithoutTimestamp(properties, file, getComment());
+        } catch (FileNotFoundException e) {
+            getLog().error("Could not create FileOutputStream: " + file);
+            throw new MojoExecutionException(e.getMessage(), e);
+        } catch (IOException e) {
+            getLog().error("Error writing properties: " + file);
+            throw new MojoExecutionException(e.getMessage(), e);
         }
     }
 
     // https://github.com/apache/maven-archiver/blob/master/src/main/java/org/apache/maven/archiver/PomPropertiesUtil.java#L81
-    private void storeWithoutTimestamp( Properties properties, File outputFile, String comments )
-        throws IOException
-    {
-        try ( PrintWriter pw = new PrintWriter( outputFile, "ISO-8859-1" ); StringWriter sw = new StringWriter() )
-        {
-            properties.store( sw, comments );
-            comments = '#' + comments;
+    private void storeWithoutTimestamp(Properties properties, File outputFile, String comment) throws IOException {
+        try (PrintWriter pw = new PrintWriter(outputFile, StandardCharsets.ISO_8859_1); StringWriter sw = new StringWriter()) {
+            properties.store(sw, comment);
 
             List<String> lines = new ArrayList<>();
-            try ( BufferedReader r = new BufferedReader( new StringReader( sw.toString() ) ) )
-            {
-                String line;
-                while ( ( line = r.readLine() ) != null )
-                {
-                    if ( !line.startsWith( "#" ) || line.equals( comments ) )
-                    {
-                        lines.add( line );
-                    }
-                }
+            try (BufferedReader r = new BufferedReader(new StringReader(sw.toString()))) {
+                r.lines()
+                        .forEach(line -> {
+                            if (line.equals('#' + comment)) {
+                                pw.println(line);
+                            }
+                            if (!line.startsWith("#")) {
+                                lines.add(line);
+                            }
+                        });
```

Comment:
Guys, to be honest I do not understand how that used to work in the original code at all. Even in original code there were a lot of mess, at least from my standpoint:

```
if ( !line.startsWith( "#" ) || line.equals( comments ) )
```

that does not seem to make any sense, because we will write the line if it is not a comment, which is perfectly reasonable, but the second clause is really strange. Espetially given the fact the value of `comments` is always `#Properties`. That does not make sense to me.

I have a couple of similar questions/proposals to your code as well:

1. Do you really think that this chain: writing properties firstly in `StringWriter`, then reading it back into `StringReader`, and then sorting it, and that writing to disk is no redundant? At very least we can do the following : 

- we can just sort the properties in memory first, 
- then select the ones that we want to write - and, honestly, I think it would be all properties, I do not think that there should be some filtering
- then just flush it to disk. 

So, there is a whole hell of this `java.io` classes and they do not make any sense. 

2. Also this `comments` field - I kinda understand what it _supposed_ to mean, but I really do not think it is of any use at all. Drop it please.

3. Sorting can be useful, I guess, but I think you implemented it wrong. If the `sort` parameter is false, _**then you sort anyway**_ but in reverse to natural order. Let's just stick to the following: is sort parameter is provided and is true, then sort naturally. Otherwise - do not sort anything.

Reply:
Thank you for your question, Mikhail.
1. Honestly, i just support logic that was in this place and i added different comment which user want. But i think you was right and we can just Drop it :) because it is useles.
In other on other suggestions i will make improvements thanks
2. Okay, I'll drop it
3. true/false i means that true is ASC, false is DESC. Do you think that true is ASC and false is not sort will be better?

Reply:
I do not think making so a good idea.

If you really want to include sorting, just make `sort` parameter not a boolean but a string, and later parse it as enum with values as `ASC` and `DESC`. 

The problem is that when sort is false in your logic, you still sort, but in descending order. It is controversial.

Reply:
Yes, you're right. I thought about it but i thought that boolean will be more useful :) but yes, enum is more useful as a result. 
Also i think that `DESC` sort will used hardly and really `true` - turn on sorting, `false` - turn off sorting will be more flexible. I'll fixed it